### PR TITLE
Close leaked I/O stream and remove unused codes

### PR DESCRIPTION
**Pull request by kezhenxu94**
### Motivation:

Clean up unused codes and migrate to JDK8.

### Modifications:

- Remove unused codes.

- Close leaked I/O stream.

- Remove meaningless comments.

- Other miscellaneous changes.

### Result:

- No unnecessary codes concerns.

- No resources leak.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Some wrong change? All tests are falling.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/loader/AgentClassLoader.java [hunk 1]
```diff
@@ -62,34 +61,22 @@
...
-                try {
-                    method.setAccessible(true);
-                    method.invoke(null);
-                } catch (Exception e) {
-                    logger.warn(e, "can not invoke ClassLoader.registerAsParallelCapable()");
-                }
-                return;
-            }
-        }
+        registerAsParallelCapable();
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/loader/AgentClassLoader.java]: You should delete this method, and move `registerAsParallelCapable` to the static initial block.

**Comment by arugal** [comment]
This is also unused code.
https://github.com/apache/skywalking/blob/e44bc36281cc0c79f263906464c62698bbf5959d/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/TraceSegmentServiceClient.java#L48

**Comment by kezhenxu94** [comment]
> This is also unused code.
> 
> https://github.com/apache/skywalking/blob/e44bc36281cc0c79f263906464c62698bbf5959d/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/TraceSegmentServiceClient.java#L48

I don't know whether you've read the changed files or not, it's already removed in the very first commit in this PR

**Pull request merged**